src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -184,10 +185,14 @@ struct SceneTableData : public SceneTableEntry, PersistentData<kPersistentSceneB
         }
         // Empty name will be initialized if the name wasn't stored
         mStorageData.SetName(nameSpan);
-
         ReturnErrorOnFailure(reader.Get(mStorageData.mSceneTransitionTimeMs));
-        ReturnErrorOnFailure(
-            mStorageData.mExtensionFieldSets.Deserialize(reader, TLV::ContextTag(TagScene::kExtensionFieldSetsContainer)));
+
+        CHIP_ERROR err = reader.Next();
+        if (CHIP_END_OF_TLV != err)
+        {
+            VerifyOrReturnError(TLV::kTLVType_Array == reader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
+            ReturnErrorOnFailure(mStorageData.mExtensionFieldSets.Deserialize(reader));
+        }
```

Comment:
This setup means that if we ever add a field _after_ the array, the array will always have to be encoded....  This should ideally do some more dynamic checking on what the actual tag here is before claiming that we are looking at extension fieldsets.

Reply:
I came to the conclusion that it would be better to just keep storing the empty EFS.